signac/core/h5store.py

```diff
@@ -215,17 +215,30 @@ def __eq__(self, other):
 
 
 class H5Store(MutableMapping):
-    """An HDF5 backend for storing arbitrary array-like and dictionary-like data.
-    Both attribute-based and index-based operations are supported.
+    """An HDF5-backed container for storing array-like and dictionary-like data.
+
+    The H5Store API implements the MutableMapping API and behaves similar
```

Comment:
We include the Python docs in signac via intersphinx, so it might be nice to have an intersphinx link to `collections.abc.MutableMapping` here.